Shopify/shopify-cli
# Fix version check for Windows

### WHY are these changes introduced?

Bugsnag issue: https://app.bugsnag.com/shopify/shopify-cli/errors/6267003737a0db00087bd914

Since v2.16.0 (https://github.com/Shopify/shopify-cli/pull/2189), we are checking for new versions in the background by using `fork`, but it's not supported on some Windows environments.

### WHAT is this pull request doing?

Only use fork when it's available, and run the check inline otherwise.

### How to test your changes?

1. Enable [these lines](https://github.com/Shopify/shopify-cli/blob/cb0df573adb1aa9c07347d9401dfd2d974e12a60/lib/shopify_cli/core/entry_point.rb#L13-L14) in your dev environment (need to play with the `if/elsif` or just copy the lines to after the conditional)
2. Change the [version](https://github.com/Shopify/shopify-cli/blob/cb0df573adb1aa9c07347d9401dfd2d974e12a60/lib/shopify_cli/version.rb#L2) to something older
3. Change the [checking interval](https://github.com/Shopify/shopify-cli/blob/cb0df573adb1aa9c07347d9401dfd2d974e12a60/lib/shopify_cli/context.rb#L19) to 1 second so it'll check each time you run the command
4. Run `bin/shopify version` a few times and you'll start to see the new version warning (hopefully on round 2)

### Update checklist
